src/test/java/se/sundsvall/relations/api/RelationResourceTest.java

```diff
@@ -0,0 +1,277 @@
+		webTestClient.delete()
+			.uri(uriBuilder -> uriBuilder.path("/{municipalityId}/relation/{id}").build(Map.of("municipalityId", MUNICIPALITY_ID, "id", RELATION_ID)))
+			.exchange()
+			.expectStatus().isNoContent()
+			.expectHeader().contentType(ALL_VALUE)
+			.expectBody().isEmpty();
+
+		verify(serviceMock).deleteRelation(MUNICIPALITY_ID, RELATION_ID);
+	}
+
+	@ParameterizedTest
+	@MethodSource("allMethodsAndUriWithRequestBody")
+	void validateReadOnly(String httpType, Function<UriBuilder, URI> uriFunction) {
+
+		final var relationInstance = createRelationInstance();
+
+		relationInstance.setCreated(OffsetDateTime.now());
+		relationInstance.setModified(OffsetDateTime.now());
+		relationInstance.setId("id");
+
+		final var response = webTestClient.method(HttpMethod.valueOf(httpType))
+			.uri(uriFunction)
+			.bodyValue(relationInstance)
+			.exchange()
+			.expectStatus().isBadRequest()
+			.expectBody(ConstraintViolationProblem.class)
+			.returnResult();
+
+		assertThat(response.getResponseBody()).isNotNull();
+		assertThat(response.getResponseBody().getTitle()).isEqualTo("Constraint Violation");
+		assertThat(response.getResponseBody().getViolations())
+			.hasSize(3)
+			.extracting("field", "message").containsExactlyInAnyOrder(
+				Tuple.tuple("id", "must be null"),
+				Tuple.tuple("created", "must be null"),
+				Tuple.tuple("modified", "must be null"));
+	}
+
+	private static Stream<Arguments> allMethodsAndUriWithRequestBody() {
+		Function<UriBuilder, URI> relationUri = uriBuilder -> uriBuilder.path("/{municipalityId}/relation").build(Map.of("municipalityId", MUNICIPALITY_ID));
+		Function<UriBuilder, URI> relationUriWithId = uriBuilder -> uriBuilder.path("/{municipalityId}/relation/{id}").build(Map.of("municipalityId", MUNICIPALITY_ID, "id", "some-id"));
+
+		return Stream.of(
+			Arguments.of("POST", relationUri),
+			Arguments.of("PUT", relationUriWithId));
+	}
+
+	@ParameterizedTest
+	@MethodSource("allMethodsAndUri")
+	void validateMunicipalityId(String httpType, Function<UriBuilder, URI> uriFunction) {
+		final var response = webTestClient.method(HttpMethod.valueOf(httpType))
+			.uri(uriFunction)
+			.bodyValue(createRelationInstance())
+			.exchange()
+			.expectStatus().isBadRequest()
+			.expectBody(ConstraintViolationProblem.class)
+			.returnResult();
+
+		assertThat(response.getResponseBody()).isNotNull();
+		assertThat(response.getResponseBody().getTitle()).isEqualTo("Constraint Violation");
+		assertThat(response.getResponseBody().getViolations())
+			.hasSize(1).first()
+			.extracting("message").isEqualTo("not a valid municipality ID");
+	}
+
+	private static Stream<Arguments> allMethodsAndUri() {
+		Function<UriBuilder, URI> relationUri = uriBuilder -> uriBuilder.path("/{municipalityId}/relation").build(Map.of("municipalityId", "1234"));
+		Function<UriBuilder, URI> relationUriWithId = uriBuilder -> uriBuilder.path("/{municipalityId}/relation/{id}").build(Map.of("municipalityId", "1234", "id", RELATION_ID));
```

Comment:
Change to use a non valid relation id here and verify that `@ValidUuid` is triggered?